src/SFA.DAS.EAS.Portal/Application/Commands/Account/AccountCreatedCommandHandler.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using SFA.DAS.EAS.Portal.Application.Services;
+using SFA.DAS.EAS.Portal.Client.Database.Models;
+
+namespace SFA.DAS.EAS.Portal.Application.Commands.Account
+{
+    public class AccountCreatedCommandHandler : ICommandHandler<AccountCreatedCommand>
+    {
+        private readonly IAccountDocumentService _accountsService;
+
+        public AccountCreatedCommandHandler(IAccountDocumentService accountsService)
+        {
+            _accountsService = accountsService;            
+        }
+
+        public async Task Handle(AccountCreatedCommand command, CancellationToken cancellationToken = default)
+        {
+            var accountDocument = await _accountsService.Get(command.Id, cancellationToken);
+
+            if(accountDocument != null)
+            {
+                return;  // already handled 
+            }
+
+            accountDocument = AccountDocument.Create(command.Id);
+            accountDocument.Account.Name = command.Name;
```

Comment:
We shouldn't be storing the name of the legal entity that happens to be in the first event we process for an account, as the account name. The front end has already been fixed to not use this property.